server/api/initial_data.py

```diff
@@ -821,6 +824,49 @@ def _delete_state_file():
         pass
 
 
+def _perform_version_6_data_migrations(
+    db: server.api.db.sqldb.db.SQLDB, db_session: sqlalchemy.orm.Session
+):
+    _migrate_alert_templates_table(db, db_session)
+
+
+def _migrate_alert_templates_table(
+    db: server.api.db.sqldb.db.SQLDB, db_session: sqlalchemy.orm.Session
+):
+    pre_defined_templates = [
+        mlrun.common.schemas.AlertTemplate(
+            template_name="JobFailed",
+            template_description="Generic template for job failure alerts",
+            system_generated=True,
+            description="A job has failed",
+            severity=mlrun.common.schemas.alert.AlertSeverity.MEDIUM,
+            trigger={"events": [mlrun.common.schemas.alert.EventKind.FAILED]},
+            reset_policy=mlrun.common.schemas.alert.ResetPolicy.MANUAL,
+        ),
+        mlrun.common.schemas.AlertTemplate(
+            template_name="DriftDetected",
+            template_description="Generic template for drift detected alerts",
+            system_generated=True,
+            description="A model drift has been detected",
+            severity=mlrun.common.schemas.alert.AlertSeverity.HIGH,
+            trigger={"events": [mlrun.common.schemas.alert.EventKind.DRIFT_DETECTED]},
+            reset_policy=mlrun.common.schemas.alert.ResetPolicy.MANUAL,
+        ),
+        mlrun.common.schemas.AlertTemplate(
+            template_name="DriftSuspected",
+            template_description="Generic template for drift suspected alerts",
+            system_generated=True,
+            description="A model drift is suspected",
+            severity=mlrun.common.schemas.alert.AlertSeverity.MEDIUM,
+            trigger={"events": [mlrun.common.schemas.alert.EventKind.DRIFT_SUSPECTED]},
+            reset_policy=mlrun.common.schemas.alert.ResetPolicy.MANUAL,
+        ),
```

Comment:
I'm not sure I like using the data migration mechanism in order to ensure predefined data in the DB. That isn't what data migrations is meant to do. This means in order to change them in the future (if needed) you'll need to add more data migrations and this will become messy fast.
Wondering if maybe worth saving these predefined templates as part of mlrun config (Good for customizing them easily) and when api initializes in `initial_data` but not as part of a "migration" we can check the DB and add the default templates to the table overriding them (only override the predefined templates)

Reply:
@quaark - I half-agree - I certainly agree the default templates should indeed be in the configuration.
These should be added to the DB as part of the `update_default_configuration_data` method (which resides in `initial_data.py`) - this is the same thing we do for the common hub target.
Note that `update_default_configuration_data` is actually not called as part of the data-migration, but rather called from `move_api_to_online`, so I actually think that's pretty much aligned with what you're looking for.